### PR TITLE
http: inject NioEventLoopGroup from transport

### PR DESCRIPTION
**Pull request by vivekrsharma**
The NioEventLoopGroup is injected by inbounds and outbounds and this can lead to a very large # of NioEventLoopGroup's very quickly. Injecting it in transport to ensure that the application has only two NioEventLoopGroup and are shared across various inbounds and outbounds.

### CODE REVIEW

File: yarpc-transport-http/src/main/java/io/yarpc/transport/http/HttpOutbound.java [hunk 1]
```diff
@@ -66,12 +66,14 @@
...
 
-  private NioEventLoopGroup workerGroup;
+  @NonNull private NioEventLoopGroup workerGroup;
+
   private Bootstrap bootstrap;
 
   HttpOutbound(@NonNull HttpTransport transport, @NonNull URL url) {
     this.transport = transport;
     this.url = url;
+    this.workerGroup = transport.workerGroup;
```
Comment (abhinav) [inline yarpc-transport-http/src/main/java/io/yarpc/transport/http/HttpOutbound.java]: inject as a constructor argument or using `transport.getWorkerGroup()`

don't directly access attributes of the transport

File: yarpc-transport-http/src/main/java/io/yarpc/transport/http/HttpTransport.java [hunk 2]
```diff
@@ -40,6 +41,9 @@
 
   private final AtomicLifecycle lifecycle = new AtomicLifecycle();
 
+  @NonNull final NioEventLoopGroup workerGroup = new NioEventLoopGroup();
+  @NonNull final NioEventLoopGroup bossGroup = new NioEventLoopGroup();
```
Comment (abhinav) [inline yarpc-transport-http/src/main/java/io/yarpc/transport/http/HttpTransport.java]: make these attributes private and possibly add getters

File: yarpc-transport-http/src/main/java/io/yarpc/transport/http/HttpOutbound.java (hunk 1 above)
Comment (vivekrsharma) [inline yarpc-transport-http/src/main/java/io/yarpc/transport/http/HttpOutbound.java]: Yes, agreed and fixed both.

File: yarpc-transport-http/src/main/java/io/yarpc/transport/http/HttpOutbound.java [hunk 3]
```diff
@@ -66,12 +66,14 @@
   /** Configuration for the HttpTransportEncoder. */
   @NonNull private final TransportRequestEncoderConfiguration requestEncoderConfig;
 
-  private NioEventLoopGroup workerGroup;
+  @NonNull private NioEventLoopGroup workerGroup;
```
Comment (abhinav) [inline yarpc-transport-http/src/main/java/io/yarpc/transport/http/HttpOutbound.java]: final?

File: yarpc-transport-http/src/main/java/io/yarpc/transport/http/HttpInbound.java [hunk 4]
```diff
@@ -68,8 +68,10 @@
    */
   @Getter private int port;
 
-  private NioEventLoopGroup bossGroup;
-  private NioEventLoopGroup workerGroup;
+  @NonNull private NioEventLoopGroup bossGroup;
+
+  @NonNull private NioEventLoopGroup workerGroup;
```
Comment (abhinav) [inline yarpc-transport-http/src/main/java/io/yarpc/transport/http/HttpInbound.java]: final?

File: yarpc-transport-http/src/main/java/io/yarpc/transport/http/HttpOutbound.java [hunk 5]
```diff
@@ -97,9 +99,7 @@ public void start() {
   }
 
   private void doStart() {
-    // TODO: Separate event loops for each outbound/inbound seems wrong.
-    // The Transport should inject this.
-    this.workerGroup = new NioEventLoopGroup();
+
```
Comment (abhinav) [inline yarpc-transport-http/src/main/java/io/yarpc/transport/http/HttpOutbound.java]: nit: delete empty line

**Comment by breerly** [comment]
What's with the coveralls failing?

**Comment by breerly** [comment]
@vivekrsharma i understand what you're doing here, just not why you are doing it.

Can you update the PR description to include the motivation for the change, so that others can more easily review it?

**Comment by bpon** [comment]
@breerly I think the coveralls threshold is set to 90%.  We're still working on it (#129).

**Comment by breerly** [comment]
@bpon seems like we should set it to 60% for now so that builds pass, and then keep raising it as the coverage increases... or is there some advantage to this approach?

**Comment by vivekrsharma** [comment]
@breerly So spoke to abg and looks like we would want the coverage to be 90% and just ignore the build failure due to incomplete coverage for now.

**Pull request merged**